jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/codeAction/proposal/quickfix/InsertAnnotationQuickFix.java

```diff
@@ -110,10 +110,23 @@ private void addAttribute(Diagnostic diagnostic, JavaCodeActionContext context,
     }
 
     protected IBinding getBinding(ASTNode node) {
+        // handle annotation insertions for a single variable declaration
+        if (node.getParent() instanceof SingleVariableDeclaration) {
+            return ((SingleVariableDeclaration) node.getParent()).resolveBinding();
+        }
+        
         if (node.getParent() instanceof VariableDeclarationFragment) {
             return ((VariableDeclarationFragment) node.getParent()).resolveBinding();
         }
         return Bindings.getBindingOfParentType(node);
     }
 
+    protected String getLabel(String annotation, String... attributes) {
+        String type = "";
+
+        String name = "Add " + attributes[0] + " to " + annotation;
+        name = name.concat(type);
+        return name;
+    }
```

Comment:
Ideally I think we should handle the two label cases in this class (adding an annotation & adding an attribution to an annotation).
What do you think about this label being "Insert `@Annotation`" and then we create a new class "InsertAnnotationAttributesQuickFix" which extends "InsertAnnotationQuickFix" and overrides the label to the above? Then we don't have to include the label specific logic in "AddPathParamsQuickFix" and it can easily be reused.

Reply:
Sounds like a good idea! will implement that!